### PR TITLE
Fix for CF-10 : keystone exception handling

### PR DESCRIPTION
**Pull request by raiesmh08**
Sometimes during authorization process we can get some exception,
raised by some temporary conditions: network instability, auth-backend
unavailability and others not related to wrong credentials.
When this encountered, cloudferry lack handling of such situations.

This patch provides fix for above situations.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:200: [W0221(arguments-differ), 
KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' 
method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/identity/keystone.py [hunk 1]
```diff
@@ -219,12 +219,15 @@ def get_client(self):
...
-                                      insecure=self.config.cloud.insecure)
+        proxy_client_by_creds = self.proxy(self._get_client_by_creds,
+                                           self.config)
+        proxy_ks_client = self.proxy(keystone_client.Client, self.config)
+
+        with proxy_client.expect_exception(ks_exceptions.Unauthorized):
+            return proxy_ks_client(auth_ref=proxy_client_by_creds().auth_ref,
+                                   endpoint=self.config.cloud.auth_url,
+                                   cacert=self.config.cloud.cacert,
+                                   insecure=self.config.cloud.insecure)
```
Comment (antonf) [inline cloudferrylib/os/identity/keystone.py]: Please use cloudferrylib.utils.retrying.Retry

Comment (SVilgelm) [inline cloudferrylib/os/identity/keystone.py]: You can wrap the get_client function with retry decorator:
```python
from cloudferrylib.utils import retrying

...

@retrying.retry(expected_exceptions=[ks_exceptions.Unauthorized], reraise_original_exception=True, max_attempts=CONF.migrate.retry)
def get_client(...):
    ...
```

Comment (raiesmh08) [inline cloudferrylib/os/identity/keystone.py]: Will do it in next patch.

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:200: [W0221(arguments-differ), 
KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' 
method
cloudferrylib/os/identity/keystone.py:216: [E0602(undefined-variable), 
KeystoneIdentity.get_client] Undefined variable 'retrying'
cloudferrylib/os/identity/keystone.py:218: [E0602(undefined-variable), 
KeystoneIdentity.get_client] Undefined variable 'CONF'

**Comment by SVilgelm** [comment]
:-1: Please write the unit tests for this case

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2741/

**Comment by SVilgelm** [comment]
Also, please merge the commits (man git-rebase  ->  squash)

**Comment by raiesmh08** [comment]
ohh yes, I will update new patch.

**Comment by raiesmh08** [comment]
Guys, I was facing some issue in squash, I need more practice on it.
For the time, I am proposed fix here - 
https://github.com/MirantisWorkloadMobility/CloudFerry/pull/877

**Pull request closed without merging**